File: core/src/main/java/hello/core/discount/implement/RateDiscountPolicy.java
package hello.core.discount.implement;

import hello.core.discount.DiscountPolicy;
import hello.core.member.data.Grade;
import hello.core.member.data.Member;

public class RateDiscountPolicy implements DiscountPolicy {

    private int discountPercent = 10;


    @Override
    public int discount(Member member, int price) {

        int testPrice = price * discountPercent / 100 ;

        if (member.getGrade() == Grade.VIP) {
            return (price * discountPercent) / 100;
        } else {
            return 0;
        }
    }
}

File: core-kotlin/src/main/kotlin/hello/corekotlin/discount/implement/RateDiscountPolicy.kt
package hello.corekotlin.discount.implement

import hello.corekotlin.discount.DiscountPolicy
import hello.corekotlin.member.data.Grade
import hello.corekotlin.member.data.Member

class RateDiscountPolicy: DiscountPolicy {

    private val dicountRate = 10;

    override fun discount(member: Member, price: Int): Int {
        return member.run {
            if (member.grade == Grade.VIP) {
                price * dicountRate / 100
            } else {
                0
            }
        }
    }
}